great_expectations/experimental/datasources/file_path_data_asset.py

```diff
@@ -78,11 +84,18 @@ def __init__(self, **data):
         )
         self._all_group_names = self._regex_parser.get_all_group_names()
 
+    @pydantic.validator("regex", pre=True)
+    def _parse_regex_string(
+        cls, regex: Optional[Union[re.Pattern, str]] = None
+    ) -> re.Pattern:
+        return Datasource.parse_regex_string(regex=regex)
```

Comment:
In looking for a way to get type hints at the API level for `regex` and `order_by` I learned that the creators of `pydantic` [recommend that you do not perform type conversion in validators](https://github.com/pydantic/pydantic/discussions/3997), but instead do the type conversion outside of the `BaseModel`. This lead me to move the logic for `parse_regex_string` into a static method that can be leveraged by the `Datasource` `add_*` methods. We do not need to keep this logic in the validator. The side effect of removing it would just be that you have to pass the actual type if you don't want to use `type: ignore`.

Reply:
@Kilo59 Could you please review this reasoning?  To me, validator should only validate and not be utilized for "post init" type purposes; but I do not have enough experience with it in practice to be certain.  Thank you.